### PR TITLE
fix enabled banner on featured experiment

### LINKED ISSUE #3272: [Featured Experiment] Agreement string with hyperlinks is not removed after enabling Notes featured experiment
**[Affected versions]:**
- Firefox v57.0.4 and above

**[Affected Platforms]:**
- All Windows
- All Mac OS
- All Linux

**[Prerequisites]:**
- Following 5 prefs have been set or added on a new profile: "extensions.legacy.enabled = true" "xpinstall.signatures.required = false" "testpilot.env = dev" "extensions.webapi.testing = true" "extensions.install.requireBuiltInCerts = false".
- Test Pilot add-on was already installed on the same profile.

**[Steps to reproduce]:**
1. Open Firefox browser with the profile from prerequisites and navigate to https://testpilot.dev.mozaws.net/experiments page.
2. Click the "Enable Notes" green button and follow the process to enable it.
3. Observe how the "Featured Experiment" section changes after Notes is enabled.

**[Expected result]:**
- The "Enable Notes" button is replaced by "Manage" gray button and "Feedback" blue one.
- The "By proceeding, you agree to the terms and privacy of Test Pilot and Notes." string is removed since experiment is enabled.
- The "Enabled" label is added on top of the experiment video presentation.

**[Actual result]:**
- The "By proceeding, you agree to the terms and privacy of Test Pilot and Notes." string is not removed after Notes experiment is enabled.

**[Notes]**
- This issue occurs also on Stage server.
- Here is a screenshot of the issue:

by proceeding not removed

### LINKED ISSUE #3374: [Featured Experiment] The "Enabled" label is no longer displayed if the experiment is enabled but Test Pilot was uninstalled
**[Notes]:**
- The only time when the label is displayed, is when the "Uninstall" flow fails and you are not redirected to "Retire" page. In that case the "Experiments" page refreshes and the label is where it needs to be. You can see it on the end of the screen recording.
- Here is a screen recording of the issue: https://goo.gl/W3Y2ZK

**[Affected versions]:**
- Firefox Nightly 60.0a1 

**[Affected Platforms]:**
- Windows
- Mac OS
- Linux

**[Prerequisites]:**
- Following 5 prefs have been set or added on a new profile: "extensions.legacy.enabled = true" "xpinstall.signatures.required = false" "testpilot.env = dev" "extensions.webapi.testing = true" "extensions.install.requireBuiltInCerts = false".

**[Steps to reproduce]:**
1. Start Firefox browser with the profile from prerequisite and navigate to https://testpilot.dev.mozaws.net page.
2. Click the "Install Test Pilot & Enable Notes" green button from "Featured Experiment" section.
3. Follow the flow and install TxP & enable Notes experiment.
4. Click on "Settings" button from top right and select "Uninstall Test Pilot" option.
5. Click the "Proceed" button on the displayed uninstall modal.
6. Click the "Home" button on the "Retire" page modal and observe the featured experiment state.

**[Expected result]:**
- Even if Test Pilot was uninstalled, the "Enabled" label is displayed on top left side of the "Featured Experiment" section above the experiment title.

**[Actual result]:**
- There is no "Enabled" label displayed on top left side above the experiment title.

### PR DESCRIPTION
**Pull request by dannycoates**
fixes #3374
fixes #3272

### CODE REVIEW

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3440/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/aa306d019fd3bc9cf998fa8ccbe6ab4cd78616c2/

File: frontend/src/app/components/MainInstallButton/index.js [hunk 1]
```diff
@@ -29,15 +29,15 @@ export default class MainInstallButton extends React.Component {
   enableExperimentWrapped() {
     const { experiment, enableExperiment, postInstallCallback } = this.props;
 
-    this.setState({ isInstalling: true }, () => enableExperiment(experiment)
+    this.setState({ isInstalling: true });
+    enableExperiment(experiment)
       .then(() => {
         if (postInstallCallback) {
           postInstallCallback();
-          this.setState({isInstalling: false});
```
Comment (dannycoates) [inline frontend/src/app/components/MainInstallButton/index.js:36]: This triggered a warning by react saying the component had already been unmounted

File: frontend/src/app/lib/InstallManager.js [hunk 2]
```diff
@@ -198,10 +198,16 @@ export function setupAddonConnection(store) {
...
   }
 
-  mam
+  dispatch(
+    updateExperiment(experiment.addon_id, {
+      inProgress: true
+    })
+  );
+
+  return mam
```
Comment (dannycoates) [inline frontend/src/app/lib/InstallManager.js:210]: The previous PR that changed this file stopped returning a promise. I didn't catch it then so I'm fixing it now.

File: frontend/src/app/reducers/addon.js [hunk 3]
```diff
@@ -84,8 +84,7 @@ function setHasAddon(
   } catch (e) {}
   return {
     ...state,
-    hasAddon,
-    installed: hasAddon ? state.installed : {}
```
Comment (dannycoates) [inline frontend/src/app/reducers/addon.js:88]: we no longer want to wipe the installed experiment list when the txp addon is removed

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3440/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/97d35e0b9069e2888f69c4e437232ea559e2ab96/

**Pull request merged**